manufacturing-client/src/main.rs

```diff
@@ -240,130 +309,164 @@ async fn main() -> Result<()> {
         return Ok(CborSimpleType::Text(mfg_info.into_string().unwrap()));
     }
-    match mfg_string_type {
+    log::debug!("mfg_string_type '{mfg_string_type:?}' requested");
+    let mfg_iden = match mfg_string_type {
         MfgStringType::SerialNumber => {
-            log::debug!(
-                "mfg_string_type value:{:#?} refers SerialNumber format ",
-                mfg_string_type
-            );
-            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+            fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
                 .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
-                .context("Error determining system serial number")?;
-            Ok(CborSimpleType::Text(serial_dmi))
+                .context("Error determining system serial number")?
         }
-        MfgStringType::StructuredDeviceInfo => {
-            log::debug!(
-                "mfg_string_type value:{:#?} refers StructuredDeviceInfo format ",
-                mfg_string_type
-            );
-            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
-                .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
-                .map(|s| s.trim_end().to_owned())
-                .context("Error determining system serial number")?;
-
-            let path = "/sys/class/net/";
-            let mut serial_mac = String::new();
-
-            for entry in fs::read_dir(path)? {
-                let entry = entry?;
-                let path_mac_addr = entry.path().join("address");
-                let interface = entry.file_name().to_string_lossy().to_string();
-                let mac = fs::read_to_string(path_mac_addr)
-                    .map(|s| s.trim_end().to_owned())
-                    .context("Error reading MAC address")?;
-
-                log::debug!("MAC address read from file /sys/class/net/{interface}/address {mac}");
-
-                lazy_static! {
-                    static ref MAC_REGEX: Regex =
-                        Regex::new(r"^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$").unwrap();
-                }
-                if MAC_REGEX.is_match(&mac) {
-                    let mut non_zero_mac = false;
-                    for byte in mac.split(':') {
-                        let byte = u8::from_str_radix(byte, 16)?;
-                        if byte != 0 {
-                            non_zero_mac = true;
-                            break;
-                        }
-                    }
-                    if non_zero_mac {
-                        write!(serial_mac, "IFACE_{interface}={mac};").unwrap();
-                        log::debug!("MAC address {serial_mac} of interface {interface} ");
-                    }
-                }
+        MfgStringType::MACAddress => {
+            let given_iface = iface.context("No iface provided")?;
```

Comment:
the check on iface is done already in main() , so this check is optional ,I guess.

Reply:
yeah, but we should never use un-checked inputs on functions since the function might be called from somewhere else other than main in the future